src/framework/filterControl/FilterControl.jsx

```diff
@@ -51,94 +52,86 @@ export default class FilterControl extends Component {
     });
   }
 
-  onSelectFilterOption(filter) {
+  onSelectFilterOption(filterOption) {
     this.setState({
-      isDropdownOpen: true,
-      selectedFilterOption: filter,
+      selectedFilterOption: filterOption,
     });
   }
 
-  onToggleClick() {
+  onAddButtonClick() {
     this.setState({
-      isDropdownOpen: !this.state.isDropdownOpen,
+      isDropdownOpen: true,
       selectedFilterOption: null,
     });
   }
 
   renderDropdownContent() {
-    // If we don't have a selected filter yet, then we're in the process
-    // of selecting one.
-    if (!this.state.selectedFilterOption) {
-      return (
-        <div>
-          <div className="filterDropdown__header">
-            <div className="filterDropdown__header__title">
-              Add a Filter
-            </div>
-            <span
-              className="filterDropdown__header__closeButton"
-              onClick={this.onDropdownClose}
-            />
-          </div>
-          <FilterOptionList
-            filterOptions={this.props.filterOptions}
-            onSelectFilterOption={this.onSelectFilterOption}
-          />
-        </div>
-      );
-    }
-
-    // If we have a selected filter, then we can create a filter matcher
-    // from it.
     return (
       <div>
         <div className="filterDropdown__header">
-          <span
-            className="filterDropdown__header__backButton"
-            onClick={this.onCancelFilter}
-          />
           <div className="filterDropdown__header__title">
-            {this.state.selectedFilterOption.name}
+            Add a Filter
           </div>
+          <span
+            className="filterDropdown__header__closeButton"
+            onClick={this.onDropdownClose}
+          />
         </div>
-        <FilterForm
-          filterOption={this.state.selectedFilterOption}
-          onAddFilter={this.onAddFilter}
+        <FilterOptionList
+          filterOptions={this.props.filterOptions}
+          onSelectFilterOption={this.onSelectFilterOption}
         />
       </div>
     );
   }
 
+  renderFilterItems() {
+    const filterItems = this.props.selectedFilters.map(filter => (
+      (
+        <FilterItem
+          filter={filter}
+          key={filter.filterOption.name}
+          onRemoveSelectedFilter={this.props.onRemoveSelectedFilter}
+          onReplaceFilter={this.props.onReplaceFilter}
+        />
+      )
+    ));
+
+    return (
+      <div className="filterItemList">
+        {filterItems}
+        {this.state.selectedFilterOption !== null &&
+          <CreateFilterFormDropdown
```

Comment:
When this renders the SingleSelectFilterForm, following changes should be made to that:

1. Use primary button instead of native button.
2. Use UI framework Input component if available.
3. Most importantly, I think we do not need the label on top of the input because that is already rendered in the `renderForm()` method of the `FilterControl` and feels odd to render it twice.